File: example/ts/1001/main.ts
import * as readline from 'readline';

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.on('line', (line: string) => {
    var nums = line.split(' ');
    var a = parseInt(nums[0]);
    var b = parseInt(nums[1]);
    var res = a + b;
    console.log(res);
    process.exit(0);
});

File: example/rust/1001/main.rs
use std::io;

fn main() {
    let mut input = String::new();

    io::stdin().read_line(&mut input).expect("correct input");
    let res = input
        .trim()
        .split(' ')
        .map(|a| a.parse::<i32>())
        .map(|a| a.expect("parsed integer"))
        .fold(0i32, |sum, a| sum + a);

    println!("{}", res);
}
